plugin/pkg/scheduler/algorithm/types.go

```diff
@@ -54,10 +54,15 @@ type PriorityConfig struct {
 	Weight   int
 }
 
-func EmptyMetadataProducer(pod *api.Pod, nodeNameToInfo map[string]*schedulercache.NodeInfo) interface{} {
-	return nil
+// emptyMetadataProducer returns a no-op MetadataProducer.
```

Comment:
I don't understand why you did this change. It's more complcated now and not needed in my opinion.